tests/ObjectEncryptorTest.php

```diff
@@ -401,27 +365,24 @@ public function testEncryptorSimpleObjectEncrypted(): void
 
     public function testEncryptorNestedArray(): void
     {
-        $encryptor = $this->factory->getEncryptor();
+        $encryptor = $this->getEncryptor();
         $array = [
             'key1' => 'value1',
             'key2' => [
                 'nestedKey1' => 'value2',
                 'nestedKey2' => [
-                    '#finalKey' => 'value3'
-                ]
-            ]
+                    '#finalKey' => 'value3',
+                ],
+            ],
         ];
-        $result = $encryptor->encrypt($array, $encryptor->getRegisteredComponentWrapperClass());
-        self::assertArrayHasKey('key1', $result);
-        self::assertArrayHasKey('key2', $result);
-        self::assertArrayHasKey('nestedKey1', $result['key2']);
-        self::assertArrayHasKey('nestedKey2', $result['key2']);
-        self::assertArrayHasKey('#finalKey', $result['key2']['nestedKey2']);
+        $result = $encryptor->encryptForComponent($array, 'my-component');
+        self::assertIsArray($result);
         self::assertEquals('value1', $result['key1']);
         self::assertEquals('value2', $result['key2']['nestedKey1']);
-        self::assertStringStartsWith('KBC::ComponentSecure', $result['key2']['nestedKey2']['#finalKey']);
+        self::assertStringStartsWith('KBC::ComponentSecureKV::', $result['key2']['nestedKey2']['#finalKey']);
```

Comment:
:thinking: proc se to tu zmenilo na `ComponentSecureKV`?

Reply:
Protože předtím se to konfigurovalo ve factory jinak a byl tam první KMS wrapper

Reply:
jo jasně, jak říkáš